<a href="https://colab.research.google.com/github/Abishek46/Mini-Project-14-Cosine-Similarity/blob/main/Mini_Project_14_Cosine_Similarity.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
from sklearn.metrics.pairwise import cosine_similarity, cosine_distances

In [6]:
import pandas as pd

df = pd.DataFrame([
        {'Vivo': 3,'Oppo': 1},
        {'Vivo': 2,'Oppo': 2},
        {'Vivo': 2,'Oppo': 3},
        {'Vivo': 1,'Oppo': 2},
    ],
    index=[
        "doc1",
        "doc2",
        "doc3",
        "doc4"
    ])

df

Vivo  Oppo
doc1     3     1
doc2     2     2
doc3     2     3
doc4     1     2

In [7]:
cosine_similarity(df.loc["doc1":"doc1"],df.loc["doc2":"doc2"])

array([[0.89442719]])

it shows that doc1 is 89% similar than that of doc2 that is oppo